### PR TITLE
More typing!

### PR DESCRIPTION
**Pull request by abidlabs**
Done:
* `mix.py` 
* `networking.py`
* `pipelines.py`
* `processing_utils.py`
* `reload.py`
* `strings.py`
* `tunneling.py`
* `queueing.py`
* `serializing.py`
* `routes.py`

### CODE REVIEW

File: gradio/mix.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def __init__(self, *interfaces: gradio.Interface, **options):
         Returns:
             an Interface object comparing the given models
         """
-        outputs: List[IOComponent] = []
+        outputs: List[Component] = []
```
Comment (aliabid94) [inline gradio/mix.py:35]: It does have to be an IOComponent, right?

File: gradio/mix.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def __init__(self, *interfaces: gradio.Interface, **options):
 
         async def parallel_fn(*args):
             return_values_with_durations = await asyncio.gather(
-                *[interface.call_function(0, args) for interface in interfaces]
+                *[interface.call_function(0, list(args)) for interface in interfaces]
```
Comment (aliabid94) [inline gradio/mix.py:47]: why recast as list, isn't it already a list? calling list is making a shallow copy for no reason?

File: gradio/routes.py [hunk 3]
```diff
@@ -379,61 +375,60 @@ async def predict(
 
             # If this fn_index cancels jobs, then the only input we need is the
             # current session hash
-            if app.blocks.dependencies[body.fn_index]["cancels"]:
+            if app.get_blocks().dependencies[body.fn_index]["cancels"]:
                 body.data = [body.session_hash]
             if body.request:
                 if body.batched:
-                    request = [Request(**req) for req in body.request]
+                    request_ = [Request(**req) for req in body.request]
```
Comment (aliabid94) [inline gradio/routes.py:382]: I've always seen the underscore come first when avoiding another namespace, e.g. `_id` instead of `id`.

**Review by aliabid94 — APPROVED** [review]
lgtm

File: gradio/mix.py (hunk 1 above)
Comment (abidlabs) [reply gradio/mix.py:35]: Yeah I'm actually going to just remove this type hint

File: gradio/mix.py (hunk 2 above)
Comment (abidlabs) [reply gradio/mix.py:47]: It's actually a tuple not a list :)

File: gradio/routes.py (hunk 3 above)
Comment (abidlabs) [reply gradio/routes.py:382]: Sure thanks

**Comment by abidlabs** [comment]
Thanks for all of the reviews @aliabid94!

**Pull request merged**